backend/src/main/java/com/redhat/cloud/notifications/routers/internal/InternalPermissionService.java

```diff
@@ -0,0 +1,93 @@
+package com.redhat.cloud.notifications.routers.internal;
+
+import com.redhat.cloud.notifications.auth.ConsoleIdentityProvider;
+import com.redhat.cloud.notifications.db.InternalRoleAccessResources;
+import com.redhat.cloud.notifications.models.InternalRoleAccess;
+import com.redhat.cloud.notifications.routers.internal.models.AddAccessRequest;
+import com.redhat.cloud.notifications.routers.internal.models.InternalUserPermissions;
+import io.quarkus.security.identity.SecurityIdentity;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.redhat.cloud.notifications.Constants.API_INTERNAL;
+
+@RolesAllowed(ConsoleIdentityProvider.RBAC_INTERNAL_ADMIN)
+@Path(API_INTERNAL + "/access")
+public class InternalPermissionService {
+
+    @Inject
+    InternalRoleAccessResources internalRoleAccessResources;
+
+    @Inject
+    SecurityIdentity securityIdentity;
+
+    @GET
+    @Path("/me")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_INTERNAL_USER) // Overrides admin permission
+    public InternalUserPermissions getPermissions() {
+        InternalUserPermissions permissions = new InternalUserPermissions();
+        if (securityIdentity.hasRole(ConsoleIdentityProvider.RBAC_INTERNAL_ADMIN)) {
+            permissions.setAdmin(true);
+            return permissions;
+        }
+
+        String privateRolePrefix = InternalRoleAccess.getPrivateRolePrefix();
+
+        Set<String> roles = securityIdentity
+                .getRoles()
+                .stream()
+                .filter(s -> s.startsWith(privateRolePrefix))
+                .map(s -> s.substring(privateRolePrefix.length()))
+                .collect(Collectors.toSet());
+
+        List<InternalRoleAccess> access = internalRoleAccessResources.getByRoles(roles);
+        access.stream()
+                .map(InternalRoleAccess::getApplicationId)
+                .map(UUID::toString)
+                .forEach(permissions::addApplicationId);
+
+        return permissions;
+    }
+
+    @GET
+    @Path("/")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<InternalRoleAccess> getAccessList() {
```

Comment:
Is returning each `applicationId` enough? I mean there's not much we can do from that information alone, so maybe we could return things like the application name from this method.

Reply:
Yes that's what I meant. We'll take care of that later if needed.

Reply:
Too late, already implemented 😀.
I'm returning the displayName and the id

Reply:
Awesome!